test/t/calls/voicemail_test.exs

```diff
@@ -63,6 +63,58 @@ defmodule T.Calls.VoicemailTest do
     end
   end
 
+  describe "voicemail_listen_message/3" do
+    test "failure: caller can't listen to their own message" do
+      caller = onboarded_user()
+      called = onboarded_user()
+
+      match = insert(:match, user_id_1: caller.id, user_id_2: called.id)
+
+      assert {:ok, %Voicemail{} = voicemail, _new_match_expiration_date = nil} =
+               Calls.voicemail_save_message(caller.id, match.id, _s3_key = Ecto.UUID.generate())
+
+      refute Calls.voicemail_listen_message(caller.id, voicemail.id)
+      assert %Voicemail{listened_at: nil} = Repo.get!(Voicemail, voicemail.id)
+    end
+
+    test "failure: user outside of the match can't to listen to their messages" do
+      caller = onboarded_user()
+      called = onboarded_user()
+      spy = onboarded_user()
```

Comment:
noice

Reply:
this test actually failed in the first attempt ...